docs/mkdocs/documentation/ordered_upgrade.md

```diff
@@ -0,0 +1,71 @@
+6. customer restores its workload on the node.
+
+In case customer does not need to to execute additional upgrading actions (point 4), then all he needs to do is just to 
+change the kmm.node.kubernetes.io/version-module-<module-namespace>-<module-name>=<module version> label.
+Meaning: steps 3,4 and 5 are unified into one step which means: customer changes kmm.node.kubernetes.io/version-module-<module-namespace>-<module-name>
+label value to new <module version>, as the one set in the KMM CR
+
+## Inner implementation
+
+### Components:
+
+1. Dedicated controller that watches for kmm.node.kubernetes.io/version-module-<module-namespace>-<module-name>. Once it is notified of a change in that label,
+   it will use inner labels: beta.kmm.node.kubernetes.io/version-module-loader-<module-namespace>-<module-name> and
+   beta.kmm.node.kubernetes.io/version-device-plugin-<module-namespace>-<module-name> to initiate the upgrade process
+2. ModuleLoader and DevicePlugin daemonsets now have additional node selector value - beta.kmm.node.kubernetes.io/version-module-loader--<module-namespace>-<module-name>
+   and kmm.node.kubernetes.io/version-device-plugin-<module-namespace>-<module-name> accordingly. Manipulation of those labels in the nodes will allow the above mentioned
+   controller to remove/add pods on the node
+3. upon change of the Version field in the Module CR, a new ModuleLoader and DevicePlugin daemonsets will be created with appropriate labels in the NodeSelector field
+
+### Flow
+
+1. In case Version field was changed in ther Module CR, a new ModuleLoader and DevicePlugin daemonsets ae created with appropriate labels:
+   beta.kmm.node.kubernetes.io/version-module-loader-<module-namespace>-<module-name>=<version> and 
+   beta.kmm.node.kubernetes.io/version-device-plugin-<module-namespace>-<module-name>=<version> set in the NodeSelectors
+2. Once "version label controller" detects change/removal of the kmm.node.kubernetes.io/version-module-<module-namespace>-<module-name> label on the node,
+   it will remove beta.kmm.node.kubernetes.io/version-device-plugin-<module-namespace>-<module-name> from the node,
+   which will cause the device-plugin pod to be removed
+4. Once device plugin is removed, the controller will remove the beta.kmm.node.kubernetes.io/version-module-loader-<module-namespace>-<module-name> label,
+   which will cause the current version of the ModuleLoader pod to be removed
+5. Once the current ModuleLoader pod is removed, the controller will set beta.kmm.node.kubernetes.io/version-module-loader-<module-namespace>-<module-name>=<version>
+   on the node which will cause the new ModuleLoader pod to start on the controller. 
+6. Once the new ModuleLoader pod is running, the controller will set beta.kmm.node.kubernetes.io/version-device-plugin-<module-namespace>-<module-name>=<version> label
+   on the node, which will cause the DevicePlugin pod to start again on the node  
```

Comment:
```suggestion
1. In case the `version` field was changed in the `Module` CR, new ModuleLoader and DevicePlugin daemonsets are created with appropriate labels:
   `beta.kmm.node.kubernetes.io/version-module-loader-<module-namespace>-<module-name>=$moduleVersion` and 
   `beta.kmm.node.kubernetes.io/version-device-plugin-<module-namespace>-<module-name>=$moduleVersion` set in the `nodeSelector`s
2. Once "version label controller" detects change/removal of the `kmm.node.kubernetes.io/version-module-<module-namespace>-<module-name>` label on the node,
   it will remove `beta.kmm.node.kubernetes.io/version-device-plugin-<module-namespace>-<module-name>` from the node,
   which will cause the device-plugin pod to be removed
4. Once device plugin is removed, the controller will remove the `beta.kmm.node.kubernetes.io/version-module-loader-<module-namespace>-<module-name>` label,
   which will cause the current version of the ModuleLoader pod to be removed
5. Once the current ModuleLoader pod is removed, the controller will set `beta.kmm.node.kubernetes.io/version-module-loader-<module-namespace>-<module-name>=$moduleVersion`
   on the node which will cause the new ModuleLoader pod to start on the controller. 
6. Once the new ModuleLoader pod is running, the controller will set `beta.kmm.node.kubernetes.io/version-device-plugin-<module-namespace>-<module-name>=$moduleVersion` label
   on the node, which will cause the DevicePlugin pod to start again on the node.
```